dspace-api/src/main/java/org/dspace/storage/rdbms/SiteServiceInitializer.java

```diff
@@ -37,10 +48,19 @@ public void initializeSiteObject() {
             context.turnOffAuthorisationSystem();
             // While it's not really a formal "registry", we need to ensure the
             // default, required Groups exist in the DSpace database
+            Site site = null;
             if (siteService.findSite(context) == null) {
-                siteService.createSite(context);
+                site = siteService.createSite(context);
             }
             context.restoreAuthSystemState();
+            if (!authorizeService.authorizeActionBoolean(context, site, Constants.READ)) {
+                context.turnOffAuthorisationSystem();
+                Group anonGroup = groupService.findByName(context, Group.ANONYMOUS);
+                if (anonGroup != null) {
+                    authorizeService.addPolicy(context, site, Constants.READ, anonGroup);
+                }
+                context.restoreAuthSystemState();
+            }
```

Comment:
@tdonohue We were in the process of investigating/fixing this through the Java-based migration, but this also has the same problem as described in the initial PR description:
The site object is only created AFtER the migrations have been processed, which means that it's impossible to add in the resourcePolicies into the migrations as is.

We've investigated further, and initially, we see the following 2 options
* The ResourcePolicy for the anonymous rights to the Site object is generated in the SiteServiceInitializer **(Current implementation)**.
    * Downfall of this is that this will only be triggered on a new migration, this change by itself won’t trigger a new migration so it’s not kept up to date.
    * Requires manual database migrations to be ran
    * This will only be the case until a new migration file is added, when any new migration file is added, this will always be ran properly
* The ResourcePolicy could be generated in a java SQL class **(Latest proposed option)**
    * This will work currently as a new migration is triggered and in this new migration the ResourcePolicy will be added
    * This works immediately, doesn’t require a manual update or waiting for a new migration file to be added
    * This will NOT work for any new instances that are being set up. The migration for a new DB/DSpace instance will only run once and when the Java SQL files are being processed, the Site object doesn’t exist yet as it’ll be the first run. This causes the Site anon read to be broken for ANY new DSpaces to be setup if we go this route

Things that could work:
* Include both, which has the downside of duplicating code but it’ll always work, when making sure to add in enough checks
* Make a java flyway migration class that also creates the site if it doesn’t exist, currently the site creation is handled AFTER all the migrations, it’s done in the SiteServiceInitializer (Where the current implementation adds in the resource policies
=> Issue that this is very brittle if any migration AFTER this one alters anything to the Site table in the DB which is not desirable

Reply:
@jonas-atmire : Thank you for your thorough description of the issue here.  I see what you mean...this isn't easy to solve

Out of all those choices, it sounds like the current implementation may be our best option.  The problem (as previously stated) is that I'm not sure how we'll trigger this to run on our demo site, unless we just craft a few basic SQL statements that we can run manually to insert the missing resource policies.

The good news is though that *anyone upgrading from 5.x/6.x to 7.x* will autotrigger this callback to run.  So, this manual SQL route would only be necessary for active DSpace 7 developers.

Could you or your team craft the necessary SQL statement(s) to insert these policies manually?  If so, we can move this forward as-is and just have all DSpace 7 developers run the SQL statements manually for now.

Reply:
@tdonohue Thanks for the notes.
During the initial investigation/implementation of this, we'd already created the related query to run it through flyway. (But since that was not an option, we banked that option)

The following is the postgres version of the required query, which adds in the site wide anonymous read rights into the resourcepolicy table 
I don't have an active h2/ORACLE db running atm, so it might be easiest for someone with the appropriate setup to run this query as well to make sure that all related syntaxes are conform to the conventions. 
> insert into resourcepolicy (policy_id, resource_type_id, dspace_object, action_id, epersongroup_id ) select getnextid('resourcepolicy') as policy_id, 5 as resource_type_id, (select site.uuid from site limit 1) as dspace_object, 0 as action_id, (select epersongroup.uuid from epersongroup where name = 'Anonymous') as epersongroup_id;

Reply:
That `INSERT` works great for me. Thanks!